src/containers.js

```diff
@@ -125,6 +125,12 @@ exports.decorateTerm = (Term, { React }) => {
       if (uid === focussedSessionUid) {
         if (event.key === ENTER) {
           this.handleSearchNext();
+        } else if (event.key === ESCAPE) {
+          window.store.dispatch(toggleSearchInput(uid));
+          // set focus on term if panel was just hidden.
+          if (this.toggleInput() === false) {
```

Comment:
cool!,

I only have a little suggestion, this is processing the key-up event triggered by the visible input, so i believe you just can do:
```
  window.store.dispatch(toggleSearchInput(uid));
  if (this.props.term) this.props.term.focus();
```
makes sense or i just need more ☕️  :-)?